### PR TITLE
Fix docs to make property listing in the website docs more useful.

### LINKED ISSUE #3060: Make property listings in website docs more useful
### 🎛 Description

🙋 feature request

Currently, there are sections in the resource docs that list the properties or matchers of a resource; but it's unclear what purpose they serve.  It's not a table of contents (the names are unlinked) and it's not documentation (fully documented resources have a prose description of the property, and then at least one example).

An example is in the [apache resource](https://www.inspec.io/docs/reference/resources/apache/#properties)

* 'service', 'conf_dir', 'conf_path', 'user'

Aside from the formatting issues (fixed on #3058), that doesn't do the reader any good, aside from telling you what is there.  If that's useful, those should be links to the full property information.

In the case in which we don't have full descriptions of the properties (see [`aws_ec2_instance`](https://www.inspec.io/docs/reference/resources/aws_ec2_instance/)) I do think it is useful to have a simple property list, as an interim measure.

Briefly describe the issue and the expected behavior

### 🌍 InSpec and Platform Version

2.1.72

### 💁 Possible Solutions

Merge the "Properties" and "Property Examples" sections, into a section named "Properties".  Immediately following the section title, include a list of hyperlinked property names, which link to anchors within the page.

Consider placing such a ToC at the top of the page.

Consider automating the ToC generation.

### PR DESCRIPTION
**Pull request by Vasu1105**
Signed-off-by: Vasu1105 <vasundhara.jagdale@chef.io>

## Description

Few more resource docs updated as mentioned in the #5677 
## Related Issue

Fixes #3060
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: docs-chef-io/content/inspec/resources/mysql_conf.md [hunk 1]
```diff
@@ -48,6 +48,12 @@ where
 - `('path')` is the non-default path to the `my.cnf` file
 - `should eq 'value'` is the value that is expected
 
+## Properties
+
+This resource supports any of the settings listed in an my.cnf file as properties for e.g. max_connections
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/mysql_conf.md]:
```suggestion
This resource supports any settings listed in a `my.cnf` file as properties. For example, `max_connections`.
```

File: docs-chef-io/content/inspec/resources/opa_api.md [hunk 2]
```diff
@@ -45,6 +45,20 @@ The URL of the OPA API server.
...
+
+### result
+
+The `result` property checks whether the resource query returns an empty result.
+
+    its('result') { should be nil }
+
+### allow
+
+The `allow` property checks if specific input is as per the policy defined in OPA. If `allow` is not defined in the policy file then this matcher will not work.
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/opa_api.md]:
```suggestion
The `allow` property checks if a specific input matches the policy defined in OPA. This matcher will not work if `allow` is not defined in the policy file.
```

File: docs-chef-io/content/inspec/resources/opa_cli.md [hunk 3]
```diff
@@ -54,6 +54,20 @@ The query to be evaluated against policy and input data.
...
+
+### result
+
+The `result` property checks whether the resource query returns an empty result.
+
+    its('result') { should be nil }
+
+### allow
+
+The `allow` property checks if specific input is as per the policy defined in OPA. If `allow` is not defined in the policy file then this matcher will not work.
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/opa_cli.md]:
```suggestion
The `allow` property checks if specific input matches the policy defined in OPA. This matcher will not work if `allow` is not defined in the policy file.
```

File: docs-chef-io/content/inspec/resources/package.md [hunk 4]
```diff
@@ -36,6 +36,21 @@ where
...
 
+## Properties
+
+### version
+
+The `version` property tests if the named package version is on the system:
+
+    its('version') { should eq '1.2.3' }
+
+You can also use the `cmp OPERATOR` matcher to perform comparisons using the version attribute:
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/package.md]:
```suggestion
You can also use the `cmp` matcher to perform comparisons using the version attribute:
```

File: docs-chef-io/content/inspec/resources/packages.md [hunk 5]
```diff
@@ -31,6 +31,26 @@ A `packages` resource block declares a regular expression search to select packa
       its('statuses') { should cmp 'installed' }
     end
 
+## Properties
+
+### statuses
+
+The `statuses` property tests if packages are installed on the system
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/packages.md]:
```suggestion
The `statuses` property tests if packages are installed on the system:
```

File: docs-chef-io/content/inspec/resources/packages.md [hunk 6]
```diff
@@ -31,6 +31,26 @@ A `packages` resource block declares a regular expression search to select packa
...
+
+    its('statuses') { should cmp 'installed' }
+
+### versions
+
+The `versions` property tests the versions of the packages installed on the system
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/packages.md:44]:
```suggestion
The `versions` property tests the package versions installed on the system:
```

File: docs-chef-io/content/inspec/resources/packages.md [hunk 7]
```diff
@@ -31,6 +31,26 @@ A `packages` resource block declares a regular expression search to select packa
...
+
+    its('versions') { should cmp '3.4.0.2-4.el7' }
+
+### architectures
+
+The `architectures` property tests the architecture of packages installed on the system
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/packages.md:50]:
```suggestion
The `architectures` property tests the architecture of packages installed on the system:
```

File: docs-chef-io/content/inspec/resources/postgres_conf.md [hunk 8]
```diff
@@ -37,6 +37,12 @@ where
 - `('path')` is the non-default path to the `postgresql.conf` file (optional)
 - `should eq 'value'` is the value that is expected
 
+## Properties
+
+This resource supports any of the settings listed in an postgresql.conf file as properties for e.g. max_connections
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/postgres_conf.md:42]:
```suggestion
This resource supports any of the settings listed in a `postgresql.conf` file as properties. For example, `max_connections`:
```

File: docs-chef-io/content/inspec/resources/postgres_ident_conf.md [hunk 9]
```diff
@@ -39,13 +39,9 @@ where
...
 ## Properties
 
-    'conf_file', 'map_name', 'params', 'pg_username', 'system_username'
-
-## Property Examples
-
 ### map_name([String])
 
-`address` returns a an array of strings that matches the where condition of the filter table
+`map_name` returns a an array of strings that matches the where condition of the filter table
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/postgres_ident_conf.md:44]:
```suggestion
`map_name` returns an array of strings that matches the `where` condition of the filter table:
```

File: docs-chef-io/content/inspec/resources/security_policy.md [hunk 10]
```diff
@@ -41,6 +41,12 @@ where
 - `{ should eq 'value' }` tests the value of `policy_name` against the value declared in the test
 - `translate_sid` converts the security identifier (SID) into a human readable SID name if `true`. Default value is `false`.
 
+## Properties
+
+This resource supports any of the security policy name as properties for e.g. `SeNetworkLogonRigth`, `SeRemoteInteractiveLogonRight` etc.
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/security_policy.md:46]:
```suggestion
This resource supports any of the security policies as properties. For example, `SeNetworkLogonRight` and `SeRemoteInteractiveLogonRight`.
```

File: docs-chef-io/content/inspec/resources/sys_info.md [hunk 11]
```diff
@@ -91,12 +65,40 @@ The `short` property tests the host name cut at the first dot:
...
+      its('content') { should match sys_info.hostname }
+    end
+
+Options can be passed as arguments to hostname as well.
+
+    describe file('/path/to/some/file') do
+      its('content') { should match sys_info.hostname('full') }
+    end
+
+Currently supported arguments to `hostname` on Linux platforms are 'full'|'f'|'fqdn'|'long', 'domain'|'d', 'ip_address'|'i', and 'short'|'s'. Mac currently supports 'full'|'f'|'fqdn'|'long' and 'short'|'s'
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/sys_info.md:100]:
```suggestion
On Linux platforms, supported arguments to `hostname` are: 
- the FQDN (`full`, `f`, `fqdn`, and `long`)
- the DNS domain (`domain` or `d`)
- the network address (`ip_address` or `i`)
- the short hostname (`short` or `s`)

On macOS, supported arguments `hostname` are: 
- the FQDN (`full`, `f`, `fqdn`, and `long`)
- the short hostname (`short` or `s`)
```

File: docs-chef-io/content/inspec/resources/users.md [hunk 12]
```diff
@@ -50,128 +50,130 @@ or:
...
+
+The following examples show how to use this Chef InSpec audit resource.
+
+### Use a regular expression to find users
+
+    describe users.where { uid =~ /S\-1\-5\-21\-\d+\-\d+\-\d+\-500/ } do
+      it { should exist }
+    end
+
+### Test only allowed users exist
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/users.md]:
```suggestion
### Test that only allowed users exist:
```

File: docs-chef-io/content/inspec/resources/users.md [hunk 13]
```diff
@@ -50,128 +50,130 @@ or:
...
 
 where `0` is the count of bad passwords for a user.
 On Linux based operating systems it relies on `lastb` and for Windows it uses information stored for the user object.
 These settings will be resetted to `0` depending on your operating system configuration.
+
+## Examples
+
+The following examples show how to use this Chef InSpec audit resource.
+
+### Use a regular expression to find users
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/users.md]:
```suggestion
### Use a regular expression to find users:
```

File: docs-chef-io/content/inspec/resources/xinetd_conf.md [hunk 14]
```diff
@@ -38,6 +38,64 @@ where
...
+
+    its('services') { should include 'service_name' }
+
+### socket_types
+
+The `socket_types` property tests if a service listed under `/etc/xinet.d` is configured to use the named socket type:
+
+    its('socket_types') { should eq 'socket' }
+
+where `socket` is one of `dgram`, `raw`, or `stream`. For a UDP-based service:
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/xinetd_conf.md]:
```suggestion
The `socket_types` property tests if a service listed under `/etc/xinet.d` is configured to use the named socket type.

Use `socket` if the socket type is `dgram`, `raw`, or `stream`:

    its('socket_types') { should eq 'socket' }

For a UDP-based service:
```

File: docs-chef-io/content/inspec/resources/xinetd_conf.md [hunk 15]
```diff
@@ -38,6 +38,64 @@ where
...
+
+    its('type') { should eq 'TYPE' }
+
+where `'TYPE'` is `INTERNAL` (for a service provided by xinetd), `RPC` (for a service based on remote procedure call), or `UNLISTED` (for services not under `/etc/services` or `/etc/rpc`).
+
+### wait
+
+The `wait` property tests how a service handles incoming connections.
+
+For UDP (`dgram`) socket types the `wait` property should test for `yes`:
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/xinetd_conf.md]:
```suggestion
For UDP (`dgram`) socket types, the `wait` property should test for `yes`:
```

File: docs-chef-io/content/inspec/resources/xinetd_conf.md [hunk 16]
```diff
@@ -38,6 +38,64 @@ where
...
+
+    its('socket_types') { should eq 'dgram' }
+    its('wait') { should eq 'yes' }
+
+For TCP (`stream`) socket types the `wait` property should test for `no`:
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/xinetd_conf.md]:
```suggestion
For TCP (`stream`) socket types, the `wait` property should test for `no`:
```

**Review by IanMadd — COMMENTED** [review]
Docs feedback. Thanks for making this PR!

**Comment by Vasu1105** [comment]
Thanks, @IanMadd. Done required changes.

**Review by IanMadd — APPROVED** [review]

**Pull request merged**